### PR TITLE
Stream fix. Font fixes.

### PR DESCRIPTION
**Pull request by shem-sergey**

### CODE REVIEW

File: src/main/java/org/verapdf/as/io/ASMemoryInStream.java [hunk 1]
```diff
@@ -28,6 +31,43 @@ public ASMemoryInStream(byte[] buffer) {
...
+     *
+     * @param stream is stream to read into byte array.
+     */
+    public ASMemoryInStream(InputStream stream) throws IOException {
+        this.currentPosition = 0;
+        this.copiedBuffer = true;
+        this.buffer = new byte[0];
+        byte[] temp = new byte[ASBufferingInFilter.BF_BUFFER_SIZE];
+        int read = stream.read(temp);
+        do {
```
Comment (BezrukovM) [inline src/main/java/org/verapdf/as/io/ASMemoryInStream.java]: just while

File: src/main/java/org/verapdf/io/InternalInputStream.java [hunk 2]
```diff
@@ -130,17 +127,18 @@ private File createTempFile(InputStream input) throws IOException {
...
 		FileOutputStream output = null;
 		try {
 			File tmpFile = File.createTempFile("tmp_pdf_file", ".pdf");
 			tmpFile.deleteOnExit();
 			output = new FileOutputStream(tmpFile);
+			output.write(alreadyRead);
 
 			//copy stream content
-			byte[] buffer = new byte[ASBufferingInFilter.BF_BUFFER_SIZE];
+			byte[] buffer = new byte[4096];
```
Comment (BezrukovM) [inline src/main/java/org/verapdf/io/InternalInputStream.java]: byte[] buffer = new byte[ASBufferingInFilter.BF_BUFFER_SIZE];

File: src/main/java/org/verapdf/io/SeekableStream.java [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+    public void seekFromCurrentPosition(final long pos) throws IOException {
+        this.seek(getOffset() + pos);
+    }
+
+    public void seekFromEnd(final long pos) throws IOException {
+        final long size = this.getStreamLength();
+        this.seek(size - pos);
+    }
+
+    public byte readByte() throws IOException {
```
Comment (BezrukovM) [inline src/main/java/org/verapdf/io/SeekableStream.java]: check this

File: src/main/java/org/verapdf/io/SeekableStream.java [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+     *
+     * @param stream is stream to turn into seekable stream.
+     * @return SeekableStream that contains data of passed stream.
+     */
+    public static SeekableStream getSeekableStream(InputStream stream) throws IOException {
+        int totalRead = 0;
+        byte[] buffer = new byte[0];
+        byte[] temp = new byte[ASBufferingInFilter.BF_BUFFER_SIZE];
+        int read = stream.read(temp);
+        do {
```
Comment (BezrukovM) [inline src/main/java/org/verapdf/io/SeekableStream.java]: just while

File: src/main/java/org/verapdf/pd/font/CIDWArray.java [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+                } else if (obj.getType() == COSObjType.COS_ARRAY) {
+                    addSingleMappings(cidBegin, (COSArray) obj.getDirectBase());
+                }
+            }
+        }
+    }
+
+    private void addSingleMappings(int cidBegin, COSArray arr) {
+        for (int i = 0; i < arr.size(); i++) {
+            if(arr.at(i).getType() != COSObjType.COS_INTEGER &&
```
Comment (BezrukovM) [inline src/main/java/org/verapdf/pd/font/CIDWArray.java]: arr.at(i).getType().isNumber()

File: src/main/java/org/verapdf/pd/font/CIDWArray.java [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+ * @author Sergey Shemyakov
+ */
+public class CIDWArray {
+
+    private static final Logger LOGGER = Logger.getLogger(CIDWArray.class);
+
+    private Map<Integer, Double> singleMappings;
+    private List<CIDWArrayRange> ranges;
+
+    public CIDWArray(COSArray w) {
```
Comment (BezrukovM) [inline src/main/java/org/verapdf/pd/font/CIDWArray.java]: rewrite

File: src/main/java/org/verapdf/pd/font/truetype/TrueTypeFontParser.java [hunk 7]
```diff
@@ -85,7 +85,7 @@ void readTables() throws IOException {
         if (cmapParser != null) {
             this.cmapParser.readTable();
         } else {
-            throw new IOException("True type font doesn't contain cmap table.");
+            LOGGER.warn("True type font doesn't contain cmap table.");
```
Comment (BezrukovM) [inline src/main/java/org/verapdf/pd/font/truetype/TrueTypeFontParser.java]: debug

File: src/main/java/org/verapdf/io/SeekableStream.java (hunk 3 above)
Comment (shem-sergey) [inline src/main/java/org/verapdf/io/SeekableStream.java]: In previous implementation we threw EOFException in this situation, but it wasn't specially handled. So throwing IOException is totally fine.

**Pull request merged**